drivers/dht/dht.py

```diff
@@ -7,16 +7,21 @@
     from pyb import dht_readinto
 
 
+class DHTChecksumError(Exception):
+    pass
+
+
 class DHTBase:
-    def __init__(self, pin):
+    def __init__(self, pin, irq_block=True):
         self.pin = pin
         self.buf = bytearray(5)
+        self.irq_block = irq_block
 
     def measure(self):
         buf = self.buf
-        dht_readinto(self.pin, buf)
+        dht_readinto(self.pin, buf, self.irq_block)
```

Comment:
We could consider putting a retry loop here in case self.irq_block is false to free the users from doing that. I would avoid the infinite loop, though.. so maybe some fixed amount of retries should be introduced?

Reply:
I prefer to let the user decide manage that as he prefers. In my case, I have other things to do so I can't lock main loop for too long, so I retry later. DHT is already really slow, not worth adding more delays.

Reply:
Seems valid argument, indeed. I was just afraid that anybody has to implement this loop while using this argument anyways.